zebra-consensus/src/transaction.rs

```diff
@@ -889,11 +938,32 @@ where
 
             // # Consensus
             //
-            // > The Spend transfers and Action transfers of a transaction MUST be
-            // > consistent with its vbalanceSapling value as specified in § 4.13
-            // > ‘Balance and Binding Signature (Sapling)’ on p. 49.
+            // > The Action transfers of a transaction MUST be consistent with
+            // > its v balanceOrchard value as specified in § 4.14.
             //
-            // <https://zips.z.cash/protocol/protocol.pdf#spendsandoutputs>
+            // https://zips.z.cash/protocol/protocol.pdf#actions
+            //
+            // > [NU5 onward] If effectiveVersion ≥ 5 and nActionsOrchard > 0, then:
+            // > – let bvk^{Orchard} and SigHash be as defined in § 4.14;
+            // > – bindingSigOrchard MUST represent a valid signature under the
+            // >   transaction binding validating key bvk^{Orchard} of SigHash —
+            // >   i.e. BindingSig^{Orchard}.Validate_{bvk^{Orchard}}(SigHash, bindingSigOrchard) = 1.
+            //
+            // https://zips.z.cash/protocol/protocol.pdf#txnconsensus
+            //
+            // This is validated by the verifier. The `if` part is indirectly
+            // enforced, since the `orchard_shielded_data` is only parsed if those
+            // conditions apply in [`Transaction::zcash_deserialize`].
+            //
+            // >   As specified in § 5.4.7, validation of the 𝑅 component of the signature
+            // >   prohibits non-canonical encodings.
+            //
+            // https://zips.z.cash/protocol/protocol.pdf#txnconsensus
+            //
+            // This is validated by the verifier, inside the `redpallas` crate.
+            // It calls [`pallas::Affine::from_bytes`] to parse R and
+            // that enforces the canonical encoding.
```

Comment:
Since this implementation is in our code base I could document it there, but I decided to keep it here to match the Sapling documentation.